live/addr_test.go

```diff
@@ -0,0 +1,69 @@
+package live
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestParseChannelAddress_Valid(t *testing.T) {
+	addr := ParseChannelAddress("aaa/bbb/ccc/ddd")
+	require.True(t, addr.IsValid())
+
+	ex := ChannelAddress{
+		Scope:     "aaa",
+		Namespace: "bbb",
+		Path:      "ccc/ddd",
+	}
+
+	if diff := cmp.Diff(addr, ex); diff != "" {
+		t.Fatalf("Result mismatch (-want +got):\n%s", diff)
+	}
+}
+
+func TestParseChannelAddress_Invalid(t *testing.T) {
+	addr := ParseChannelAddress("aaa/bbb")
+	require.False(t, addr.IsValid())
+}
+
+func TestToWebSocketURL(t *testing.T) {
+	testCases := []struct {
+		desc   string
+		url    string
+		exp    string
+		expErr string
+	}{
+		{
+			desc: "Simple localhost",
+			url:  "http://localhost:3000",
+			exp:  "ws://localhost:3000/live/ws?format=protobuf",
+		},
+		{
+			desc: "With subpath",
+			url:  "http://host/with/subpath",
+			exp:  "ws://host/with/subpathlive/ws?format=protobuf",
```

Comment:
I wonder if there's a bug here, should there be a slash after "/subpath"?
```suggestion
			exp:  "ws://host/with/subpathlive/ws?format=protobuf",
```

Reply:
Good catch!